pkg/server/api/bundle/v1/service_test.go

```diff
@@ -1079,6 +1079,242 @@ func TestBatchCreateFederatedBundle(t *testing.T) {
+						logrus.ErrorKey:         "spiffeid: trust domain is empty",
+					},
+				},
+			},
+		},
+		{
+			name: "Update fails if trust domain is server trust domain",
+			bundlesToUpdate: []*types.Bundle{
+				func() *types.Bundle {
+					b := makeValidBundle(t, federatedTrustDomain)
+					b.TrustDomain = "example.org"
+					return b
+				}(),
+			},
+			expectedResults: []*bundlepb.BatchCreateFederatedBundleResponse_Result{
+				{Status: api.CreateStatus(codes.InvalidArgument, `updating a federated bundle for the server's own trust domain (example.org) s not allowed`)},
+			},
+			expectedLogMsgs: []spiretest.LogEntry{
+				{
+					Level:   logrus.ErrorLevel,
+					Message: `Invalid request: updating a federated bundle for the server's own trust domain is not allowed`,
+					Data: logrus.Fields{
+						telemetry.TrustDomainID: "example.org",
+					},
+				},
+			},
+		},
+		{
+			name:            "Update fails if bundle does not exists",
+			bundlesToUpdate: []*types.Bundle{makeValidBundle(t, federatedTrustDomain)},
+			expectedResults: []*bundlepb.BatchCreateFederatedBundleResponse_Result{
+				{
+					Status: api.CreateStatus(codes.NotFound, "bundle not found"),
+				},
+			},
+			expectedLogMsgs: []spiretest.LogEntry{
+				{
+					Level:   logrus.ErrorLevel,
+					Message: "Bundle not found",
+					Data: logrus.Fields{
+						telemetry.TrustDomainID: "another-example.org",
+						logrus.ErrorKey:         "rpc error: code = NotFound desc = no such bundle",
+					},
+				},
+			},
+		},
+		{
+			name:            "Update datastore query fails",
+			bundlesToUpdate: []*types.Bundle{makeValidBundle(t, federatedTrustDomain)},
+			dsError:         errors.New("datastore error"),
+			expectedResults: []*bundlepb.BatchCreateFederatedBundleResponse_Result{
+				{Status: api.CreateStatus(codes.Internal, `unable to update bundle: datastore error`)},
+			},
+			expectedLogMsgs: []spiretest.LogEntry{
+				{
+					Level:   logrus.ErrorLevel,
+					Message: "Unable to update bundle",
+					Data: logrus.Fields{
+						telemetry.TrustDomainID: "another-example.org",
+						logrus.ErrorKey:         "datastore error",
+					},
+				},
+			},
+		},
+		{
+			name: "Invalid bundle provided",
+			bundlesToUpdate: []*types.Bundle{
+				{
+					TrustDomain: federatedTrustDomain.String(),
+					X509Authorities: []*types.X509Certificate{
+						{
+							Asn1: []byte("malformed"),
+						},
+					},
+				},
+			},
+			expectedResults: []*bundlepb.BatchCreateFederatedBundleResponse_Result{
+				{Status: api.CreateStatus(codes.Internal, fmt.Sprintf("failed to convert bundle: unable to parse X.509 authority: %v", expectedX509Err))},
+			},
+			expectedLogMsgs: []spiretest.LogEntry{
+				{
+					Level:   logrus.ErrorLevel,
+					Message: "Failed to convert bundle",
+					Data: logrus.Fields{
+						telemetry.TrustDomainID: "another-example.org",
+						logrus.ErrorKey:         fmt.Sprintf("unable to parse X.509 authority: %v", expectedX509Err),
+					},
+				},
+			},
+		},
```

Comment:
can you next test cases:
- invalid jwt
- a case wiith multiple entries to update some with success another wiith error (to validate it is really a batch and we always return in same order).
- a test case where DS returns a "malformed" bundle?

Reply:
> invalid jwt

I believe we don't want to test that in this test file. IMO, we should only tests that the output of `ProtoToBundle()` fails for any reason (e.g: in this case a malformed X509 SVID). 

Then, the [ProtoToBundle tests](https://github.com/spiffe/spire/blob/b08409c066e798f7dae71ba9955b0aea72fefcae/pkg/server/api/bundle_test.go#L147) cover all the posible cases for failures (including the one for invalid JWT keys).

 > a case wiith multiple entries to update some with success another wiith error (to validate it is really a batch and we always return in same order).

Will do!

> a test case where DS returns a "malformed" bundle?

`BundleToProto()` [is not checking bundle "validity"](https://github.com/spiffe/spire/blob/b08409c066e798f7dae71ba9955b0aea72fefcae/pkg/server/api/bundle.go#L13) so it is not possible to test that at the moment. I wonder if it should be validating the bundle in the same way `ProtoToBundle()` does. I think it probably should. 

Probably it will be better to address this in a separate PR to also update unit tests for the other RPCs as well.